core/src/main/scala/com/rallyhealth/vapors/factfilter/evaluator/InterpretExprAsSimpleOutputFn.scala

```diff
@@ -0,0 +1,43 @@
+package com.rallyhealth.vapors.factfilter.evaluator
+
+import cats.{Eval, Foldable, Functor, Semigroupal}
+import com.rallyhealth.vapors.core.algebra.Expr
+import com.rallyhealth.vapors.factfilter.evaluator.InterpretExprAsResultFn.{Input, Output}
+
+class InterpretExprAsSimpleOutputFn[F[_] : Foldable, V, P]
+  extends VisitGenericExprWithProxyFn[F, V, P, InterpretExprAsSimpleOutputFn.GenSimpleOutput[*, P]] {
+  import InterpretExprAsSimpleOutputFn.GenSimpleOutput
+
+  override protected def visitGeneric[M[_] : Foldable, U, R](
+    expr: Expr[M, U, R, P],
+    input: Input[M, U],
+  ): GenSimpleOutput[R, P] = {
+    val result = InterpretExprAsResultFn(expr)(input)
+    (result.output, result.param :: Nil)
+  }
+}
+
+object InterpretExprAsSimpleOutputFn {
+  type GenSimpleOutput[R, P] = (Output[R], List[Eval[P]])
+
+  class SimpleOutputFnFunctorBuilder[F[_], V, P] {
+    type SimpleOutput[R] = GenSimpleOutput[R, P]
+    type SimpleOutputFn[R] = Input[F, V] => SimpleOutput[R]
+
+    implicit object SimpleOutputFunctor extends Functor[SimpleOutputFn] with Semigroupal[SimpleOutputFn] {
+      override def map[A, B](fa: SimpleOutputFn[A])(f: A => B): SimpleOutputFn[B] = fa.andThen {
+        case (o, params) => (o.copy(value = f(o.value)), params)
+      }
+
+      override def product[A, B](
+        fa: SimpleOutputFn[A],
+        fb: SimpleOutputFn[B],
+      ): SimpleOutputFn[(A, B)] = { input =>
+        val (a, aParams) = fa(input)
+        val (b, bParams) = fb(input)
+        // TODO: Product type should remove evidence if any arg has no evidence
```

Comment:
I'm ignoring this for now since we have to revisit evidence tracking anyway, but it might be worth encoding this now as a reminder. Do y'all think this makes sense? If there is no evidence for a single value of product type (think tuple or case class) then there is no evidence for the full product type instance?